src/app/loans/loans-view/loan-account-actions/add-collateral/add-collateral.component.ts

```diff
@@ -0,0 +1,70 @@
+/** Angular Imports. */
+import { Component, OnInit, Input } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+/** Custom Services. */
+import { LoansService } from 'app/loans/loans.service';
+
+/**
+ * Add Collateral component.
+ */
+@Component({
+  selector: 'mifosx-add-collateral',
+  templateUrl: './add-collateral.component.html',
+  styleUrls: ['./add-collateral.component.scss']
+})
+export class AddCollateralComponent implements OnInit {
+
+  @Input() dataObject: any;
+
+  /** Collateral form. */
+  collateralForm: FormGroup;
+  /** Loan Id. */
+  loanId: string;
+
+  /**
+   * Retrieve data from `Resolver`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {Router} router Router.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {LoansService} LoansService loans service.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private router: Router,
+              private route: ActivatedRoute,
+              private loanService: LoansService ) { }
+
+  ngOnInit() {
+    this.setCollateralForm();
+  }
+
+  /**
+   * Set Collateral form.
+   */
+  setCollateralForm() {
+    this.collateralForm = this.formBuilder.group({
+      'collateralTypeId': ['', Validators.required],
+      'value': ['', Validators.required],
+      'description': ['']
+    });
+  }
+
+  /**
+   * Submits collateral form.
+   */
+  submit() {
+    const collateralTypeId = this.collateralForm.value.collateralTypeId;
+    this.collateralForm.patchValue({
+      'collateralTypeId': collateralTypeId.toString()
+    });
```

Comment:
The `collateralTypeId` has to be number type, so you don't need to parse it to String type.